lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2231,34 +2262,45 @@ shrinkTxBody (Cardano.ShelleyTxBody e bod scripts scriptData aux val) = tail
     shrinkUpdates SNothing = []
     shrinkUpdates (SJust _) = [SNothing]
 
+
+balanceTransaction'
+    :: (UTxOIndex, Wallet (SeqState 'Mainnet ShelleyKey), Set Tx)
+    -> StdGenSeed
+    -> PartialTx
+    -> Either ErrBalanceTx SealedTx
+balanceTransaction' wal seed tx  =
+    flip evalRand (stdGenFromSeed seed) $ runExceptT $ balanceTransaction @(Rand StdGen)
+            (Ctx @(Rand StdGen) nullTracer testTxLayer)
+            (delegationAddress @'Mainnet)
+            mockProtocolParametersForBalancing
+            dummyTimeInterpreter
+            wal
+            tx
+
 -- | Tests that 'ErrAssignRedeemersUnresolvedTxIns' can in fact be returned by
 -- 'balanceTransaction'.
 prop_balanceTransactionUnresolvedInputs
     :: Wallet'
     -> ShowBuildable PartialTx
+    -> StdGenSeed
     -> Property
 prop_balanceTransactionUnresolvedInputs
     (Wallet' utxo wal pending)
-    (ShowBuildable partialTx') = withMaxSuccess 400 $
-        forAll (dropResolvedInputs partialTx') $ \(partialTx, dropped) ->
-        not (null dropped) ==>
-            forAllShow (runExceptT $ balanceTransaction
-                (Ctx nullTracer tl)
-                (delegationAddress @'Mainnet)
-                mockProtocolParametersForBalancing
-                dummyTimeInterpreter
-                (utxo, wal, pending)
-                partialTx) (show . Pretty) $ \case
-            Right _
-                -> cover 0 True "success" $ property True
-                   -- Balancing can succeed if the dropped inputs happen to be
-                   -- apart of the wallet UTxO.
-            Left (ErrBalanceTxAssignRedeemers (ErrAssignRedeemersUnresolvedTxIns _))
-                -> cover 1 True "unknown txins" True
-            Left _
-                -> property True
+    (ShowBuildable partialTx')
+    seed = withMaxSuccess 400 $
+        forAll (dropResolvedInputs partialTx') $ \(partialTx, dropped) -> do
+        not (null dropped) ==> do
+            let res = balanceTransaction' (utxo, wal, pending) seed partialTx
+            case res of
+                Right _
+                    -> cover 0 True "success" $ property True
+                       -- Balancing can succeed if the dropped inputs happen to be
+                       -- apart of the wallet UTxO.
+                Left (ErrBalanceTxAssignRedeemers (ErrAssignRedeemersUnresolvedTxIns _))
+                    -> cover 1 True "unknown txins" True
```

Comment:
Two questions:

1. Is `checkCoverage` called anywhere (and do we need to?)
2. If we really want to check that this branch is covered at least 1% of the time, then perhaps we should move the coverage check to **_outside_** of the pattern match? (Otherwise, we take the risk that this branch never matches, and therefore the check will not be made.)

Reply:
Thanks, good points! Addressed in 0574dd6409dbf30ad8fbc11f6e07bf04f2e95c0d